File: src/extras-kconfigwidgets/libkhelpclient.cpp
#include <KHelpClient>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <khelpclient.h>
#include "libkhelpclient.h"
#include "libkhelpclient.hxx"

void KHelpClient_InvokeHelp(const libqt_string param1, const libqt_string param2) {
    QString param1_QString = QString::fromUtf8(param1.data, param1.len);
    QString param2_QString = QString::fromUtf8(param2.data, param2.len);
    KHelpClient::invokeHelp(param1_QString, param2_QString);
}

File: src_original/extras-kconfigwidgets/libkhelpclient.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api.kde.org/khelpclient.html
pub const khelpclient = struct {
    /// [Qt documentation](https://api.kde.org/khelpclient.html#invokeHelp)
    ///
    /// ``` param1: []const u8, param2: []const u8 ```
    pub fn InvokeHelp(param1: []const u8, param2: []const u8) void {
        const param1_str = qtc.libqt_string{
            .len = param1.len,
            .data = param1.ptr,
        };
        const param2_str = qtc.libqt_string{
            .len = param2.len,
            .data = param2.ptr,
        };
        qtc.KHelpClient_InvokeHelp(param1_str, param2_str);
    }
};
